src/wp-includes/option.php

```diff
@@ -1133,6 +1142,94 @@ function delete_option( $option ) {
+ * and compare it to the maximum allowed size. If it ls larger, it will return 'auto-no', otherwise
+ * it will return 'auto-yes'.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param string           $option   The name of the option.
+ * @param mixed            $value    The value of the option to check its autoload value.
+ * @param string|bool|null $autoload The autoload value to check.
+ *                                   Accepts 'yes'|true to enable or 'no'|false to disable, or
+ *                                   'auto-yes' and 'auto-no' for internal purposes.
+ *                                   Any other autoload value will be forced to either `auto-yes`
+ *                                   or `auto-no`.
+ * @return string Returns the original $autoload value if valid, or 'auto-no' if the size exceeds
+ *                the maximum allowed size, or 'auto-yes' otherwise.
+ */
+function determine_option_autoload_value( $option, $value, $autoload ) {
+
+	/**
+	 * Filters the maximum size of option value in bytes.
+	 *
+	 * @param string|bool|null $autoload autoload value to check.
+	 * @param string $option name of the option.
+	 * @param string $value value of the option to check its autoload value.
+	 *
+	 * @return  string|bool|null $autoload autoload value to set. Accepted strings 'yes','no',auto-yes','auto-no' to enable
+	 *
+	 * @since 6.5.0
+	 */
+	$autoload = apply_filters( 'option_autoload_value', $autoload, $option, $value );
+
+	//check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'on' : 'off';
+	}
+
+	switch ( $autoload ) {
+		case 'yes':
+			return 'on';
+		case 'no':
+			return 'off';
+	}
+
+	/**
+	 * Filter: set the default value if null is passed as the $autoload
+	 *
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param $delfaut  null  The default autoload value.
+	 * @param $autoload string|bool|null The passed autoload value.
+	 * @param $option   string The passed option name.
+	 * @param $value    mixed The passed value to be saved.
+	 *
+	 * @retun bool|null if true returned $autoload will be set to 'auto-on' and false will set 'auto-off'.
+	 *        null will allow the size of the $autoload to be set 'auto-off' for large values.
+	 */
+
+	$autoload = apply_filters( 'wp_default_autoload_value', null, $autoload, $option, $value );
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'auto-on' : 'auto-off';
+	}
+
+	/**
+	 * Filters the maximum size of option value in bytes.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param int    $max_option_size The option-size threshold, in bytes. Default 150000.
+	 * @param string $option          The name of the option.
+	 */
+	$max_option_size = (int) apply_filters( 'wp_max_autoloaded_option_size', 150000, $option );
+	$size            = ( ! empty( $value ) ) ? strlen( $value ) : 0;
+
+	if ( $size > $max_option_size ) {
+		return 'auto-off';
+	}
+
+	return 'auto-on';
```

Comment:
See my above comment:
* Let's move this into a separate function, e.g. `wp_filter_default_autoload_value_via_option_size()`.
* This should be added as a default callback to the `wp_default_autoload_value` filter, in `default-filters.php`.
* The function should return `false` if the option is too large, or `null` otherwise.

Keep in mind that `true` should only be used if the logic explicitly determines to autoload the value which is not the case here. Such an option will still technically need to be autoloaded for now, but purely for backward compatibility.

So all that we have to do here is then return the default if no more specific one could be determined:
```suggestion
	return 'auto';
```